docs/source/best-practices/checkpointing.rst

```diff
@@ -25,25 +25,37 @@ Checkpointing is a fault tolerance technique designed to overcome the “fail-st
 Checkpointing types
 ================================
 
-Checkpointing can be implemented in different levels of your workflow:
+Implement checkpoints at different levels of your workflow:
 
-  * User-level checkpointing - suitable if you develop your own code, or have sufficient knowledge of the application code to integrate checkpointing techinques yourself. Generally, this approach is not recommended for most Discovery users.
-  * Application-level checkpointing - recommended for most Discovery users. Utilize the checkpointing tool that is already available in your software application. Most software designed for HPC have a checkpointing option, and information on proper usage is often available in the software user manual. 
-  * System-level checkpointing - done on the system side, where the state of the entire process is being saved. This option is less efficient than User-level or Application-level checkpointing as it introduces a lot of redundancy.   
+  * User-level checkpointing - suitable if you develop your code or have sufficient knowledge of the application code to integrate checkpointing techniques. We recommend this approach for some Discovery users.
+  * Application-level checkpointing - recommended for most Discovery users. Utilize the checkpointing tool that is already available in your software application. For example, most software designed for HPC has a checkpointing option, and information on proper usage is often available in the software user manual.
+  * System-level checkpointing - done on the system side, where we save the state of the entire process. This option is less efficient than User-level or Application-level checkpointing as it introduces a lot of redundancy.
+  * Model-level checkpointing - training models are often the intent of the user doing machine learning on Discovery.
+
+Which checkpointing method to use?
```

Comment:
I like this addition, but would it make more sense to maybe create a table/graphic and combine it with the last section? So we can define the checkpointing type/method, and then add the recommended method? 

Also, we should probably be consistent with the terminology and stick to either type or method just for clarity.